src/candle_components/date.rs

```diff
@@ -0,0 +1,33 @@
+use crate::{CandleComponent, CandleComponentUpdate, TakerTrade};
+
+/// This 'CandleComponent' keeps track of the opening timestamp of a Candle
+#[derive(Debug, Clone, Default)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct TimeStamp<T> {
```

Comment:
I would suggest to name this `OpeningTimestamp` as it more precisely encapsulates what it represents.
Also make sure to rename the file accordingly such that it matches the `struct` name.